### DISCUSSION TITLE
How to make multiple assertions as logical "or"?

### DISCUSSION THREAD

**Discussion by localusercamp** [pestphp/pest#1009] [General]
I can't find this in docs so...
I need to check if the value is integer or string.
I expect something similar to that:
```php
expect($value)->toBeInt()->or->toBeString();
```
I also need something like that:
```php
expect($value)->toExtendAny(['Class1', 'Class2']);
// Or
expect($value)->toExtend('Class1')->or->toExtend('Class2');
```

**Comment by owenvoke** [comment]
I don't think this is currently possible in a clean API, however you can use [`toMatchConstraint()`](https://pestphp.com/docs/expectations#expect-toMatchConstraint) to emulate this with [`LogicalOr`](https://github.com/sebastianbergmann/phpunit/blob/10.5/src/Framework/Constraint/Operator/LogicalOr.php). 🤷🏻
